pkg/driver/controller.go

```diff
@@ -196,16 +222,50 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 		return nil, status.Errorf(codes.Internal, "Failed to create Access point in File System %v : %v", accessPointsOptions.FileSystemId, err)
 	}
 
+	volContext := map[string]string{}
+
+	// Fetch mount target Ip for cross-account mount
+	if roleArn != "" {
+		mountTarget, err := localCloud.DescribeMountTargets(ctx, accessPointsOptions.FileSystemId, azName)
+		if err != nil {
+			klog.Warningf("Failed to describe mount targets for file system %v. Skip using `mounttargetip` mount option: %v", accessPointsOptions.FileSystemId, err)
+		} else {
+			volContext[MountTargetIp] = mountTarget.IPAddress
+		}
+	}
+
 	return &csi.CreateVolumeResponse{
 		Volume: &csi.Volume{
 			CapacityBytes: volSize,
 			VolumeId:      accessPointsOptions.FileSystemId + "::" + accessPointId.AccessPointId,
-			VolumeContext: map[string]string{},
+			VolumeContext: volContext,
 		},
 	}, nil
 }
 
 func (d *Driver) DeleteVolume(ctx context.Context, req *csi.DeleteVolumeRequest) (*csi.DeleteVolumeResponse, error) {
+	var (
+		localCloud cloud.Cloud
+		roleArn    string
+		err        error
+	)
+
+	// Fetch aws role ARN for cross account mount from secrets
+	secrets := req.GetSecrets()
```

Comment:
this can be refactored into function (getCloud or something) since both DeleteVol and CreateVol rely on it